src/lib/services/proxy-service.ts

```diff
@@ -42,7 +42,8 @@ export class ProxyService {
 
     private readonly services: Services;
 
-    private readonly clients: Map<ApiUser['secret'], Unleash> = new Map();
+    private readonly clients: Map<ApiUser['secret'], Promise<Unleash>> =
```

Comment:
It's weird that we store clients wrapped in promises instead of clients themselves. Why is that?

Reply:
Good question! And open to alternative solutions, but this is to store "we're processing/creating this unleash client, it might be ready soon", but we don't want to do that asynchronously as before because another request (or 15) might come while we're resolving the promise, and another client will be created. Ultimately, we want to know if someone is already processing a client for that secret, and wait until they finished before we proceed to create a new one

Reply:
The challenge is that the current frontend API creates new node SDK instances on the fly for new tokens discovered. 
Creating an SDK instance is an async operation, as it will require to fetch data to be ready.

This is probably not the best design, but fixing the bug at hand we need to make sure we await the client. I do not think there is a way to force that in to not being async, with the current implementation, where SDK instances are created on the fly. 

In an optimal implementation of this, we should not use the SDK for other than the evaluation part. This would allow us to keep the caching and async part separate from the evaluation part.